core/src/test/scala/org/genivi/sota/core/UpdateRequestSpec.scala

```diff
@@ -0,0 +1,54 @@
+package org.genivi.sota.core
+
+import akka.actor.ActorSystem
+import akka.event.Logging
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.model.Uri.Path
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import org.genivi.sota.core.data.UpdateRequest
+import org.genivi.sota.core.db.UpdateRequests
+import org.scalacheck.Gen
+import org.scalatest.{Matchers, PropSpec}
+import org.scalatest.prop.PropertyChecks
+import scala.concurrent.Future
+import slick.jdbc.JdbcBackend.Database
+
+class UpdateRequestSpec extends PropSpec with PropertyChecks with Matchers with Generators with ScalatestRouteTest {
+
+  val UpdatesPath = Path / "updates"
+
+  val databaseName = "test-database"
+
+  val db = Database.forConfig(databaseName)
+
+  override def beforeAll {
+    TestDatabase.resetDatabase( databaseName )
+  }
+
+  class Service(implicit system: ActorSystem) {
+    implicit val log = Logging(system, "updateRequestsSpec")
+    val resource = new UpdateRequestsResource(db, new UpdateService())
+  }
+
+  import UpdateRequest._
+  import spray.json.DefaultJsonProtocol._
+  import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+
+  property("Update requests can be listed")  {
+    new Service() {
+      forAll(Gen.listOf(updateRequestGen(PackageIdGen))) { (requests: Seq[UpdateRequest]) =>
```

Comment:
It should be something wrong with the test. It should not be possible to persist UpdateRequest if the package is not in db. Or am I missing smth.?